extensions/wikia/UserLogin/js/UserSignupAjaxValidation.js

```diff
@@ -22,111 +23,109 @@
 	 * @param {Event} e Browser event like input blur
 	 */
 	UserSignupAjaxValidation.prototype.validateInput = function (e) {
-		var el = $(e.target),
-			paramName = el.attr('name'),
-			params = this.getDefaultParamsForAjax();
-
-		params.field = paramName;
-		params[paramName] = el.val();
-
-		$.get(
-			wgScriptPath + '/wikia.php',
-			params,
-			this.validationHandler.bind(this, paramName)
-		);
+		var $el = $(e.target),
+			data = {},
+			inputName = $el.attr('name'),
+			value = $el.val(),
+			usernameAlias = 'userloginext01';
+
+		// don't send password values to the server, validated them here in JS
+		if (inputName === this.passwordInputName) {
+			this.validatePassword(value);
+		} else {
+			// back end expects 'userloginext01' instead of 'username'
+			if (inputName === 'username') {
+				data.field = usernameAlias;
+				data[usernameAlias] = value;
+			} else {
+				data.field = inputName;
+				data[inputName] = value;
+			}
+
+			this.sendValidationRequest(data)
+				.done(this.validationHandler.bind(this, inputName));
+		}
 	};
 
 	/**
-	 * Call back end for validation of input where user action occurred
-	 * @todo This was copied from validateInput with some added input name mapping. Clean this up in a separate ticket.
-	 * @param {Event} e Browser event like input blur
+	 * Client side validation that mirrors server side validation for password input. (SOC-316)
+	 * Validating client side reduces passwords being sent over HTTP in plain text.
+	 * @param {string} password
 	 */
-	UserSignupAjaxValidation.prototype.validateMappedInput = function (e) {
-		var el = $(e.target),
-			paramName = el.attr('name'),
-			params = this.getDefaultParamsForAjax(),
-			mappedParamName,
-			map;
-
-		// back end validation expects these fields to match the user signup form, so we'll map them to those values
-		// before sending.
-		map = {
-			'username': 'userloginext01',
-			'password': 'userloginext02'
-		};
-
-		mappedParamName = map[paramName] || paramName;
-		params.field = mappedParamName;
-		params[mappedParamName] = el.val();
-
-		$.get(
-			wgScriptPath + '/wikia.php',
-			params,
-			this.validationHandler.bind(this, paramName)
-		);
-	};
+	UserSignupAjaxValidation.prototype.validatePassword = function (password) {
+		var pwLength = password.length,
+			msg = '',
+			inputName = this.passwordInputName,
+			response = {};
+
+		// check password isn't too short (defaults to 1 char)
+		if (pwLength < wgMinimalPasswordLength) {
+			msg = mw.message('userlogin-error-wrongpasswordempty').escaped();
```

Comment:
If the minimum password length to more than 1, user gets error saying "your password is empty" rather than "your password is too short". We probably need to change this message or make a new one. Either way, it requires translation :(